arrays/arrays.c

```diff
@@ -19,45 +19,56 @@ typedef struct Array {
 /*****
  * Allocate memory for a new array
  *****/
-Array *create_array (int capacity) {
+Array *create_array(int capacity)
+{
   // Allocate memory for the Array struct
-
+  Array *newarray = malloc(sizeof(Array));
   // Set initial values for capacity and count
-
+  newarray->capacity = capacity;
+  newarray->count = 0;
   // Allocate memory for elements
+  newarray->elements = calloc(capacity, sizeof(char *));
 
+  return newarray;
 }
 
-
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
+void destroy_array(Array *arr)
+{
 
   // Free all elements
-
+  free(arr->elements);
   // Free array
-
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
-void resize_array(Array *arr) {
+void *resize_array(Array *arr)
+{
 
   // Create a new element storage with double capacity
-
+  int capacity = arr->capacity;
+  char **newelements = calloc(capacity * 2, sizeof(char *));
   // Copy elements into the new storage
-
+  for (int i = 0; i < capacity; i++)
+  {
+    newelements[i] = arr->elements[i];
+    if (newarray->elements[i] != NULL)
```

Comment:
You shouldn't need this if block in here. You're just copying the elements from the old arr->elements to the newelements array. Even if you have NULL values you can leave them alone. You want newelements to look exactly like the old arr->elements except it will have more space (i.e. more NULL spots on the end of it).